frc46_token/src/token/state.rs

```diff
@@ -627,6 +651,16 @@ pub fn decode_actor_id(key: &BytesKey) -> Option<ActorID> {
 
 impl Cbor for TokenState {}
 
+/// A summary of the current state to allow checking application specific invariants
+pub struct StateSummary<'bs, BS>
+where
+    BS: Blockstore,
+{
+    pub balance_map: BalanceMap<'bs, BS>,
+    pub allowance_map: AllowanceMap<'bs, BS>,
```

Comment:
Please fully load this state from the store so that StateSummary is self-contained and has no generics. I.e. build HashMaps here.

I have [WIP here](https://github.com/filecoin-project/builtin-actors/blob/cae57d2d67e529eeed3f51f53ab79a8e08afa22f/actors/datacap/src/testing.rs#L12-L16) building these for some testing I need, but I expect to just import this version in the future, whereupon my predicates can just use the state summary directly.